src/metrics/utils.ts

```diff
@@ -20,13 +21,18 @@ export type Options =
 export function getOrCreateMetric(
   type: Metrics,
   options: Options,
+  prometheusOptions?: PrometheusOptions,
 ): client.Metric<string> {
   const existingMetric = client.register.getSingleMetric(options.name);
 
   if (existingMetric) {
     return existingMetric;
   }
 
+  options.name = prometheusOptions?.prefix
+    ? prometheusOptions?.prefix.concat(options.name)
+    : options.name;
```

Comment:
Let's not mutate the input please. Create a new `options` object using the prefix if they provided one.